.github/workflows/integration-tests.yml

```diff
@@ -21,12 +21,22 @@ jobs:
         with:
           components: clippy, rustfmt
       
+      - name: Cache dependencies
```

Comment:
Doesn't look like the runners really persist this cache which is somewhat unfortunate, since the compilation with `--release` takes a while. But still looks tractable.